ports/stm32/qspi.c

```diff
@@ -31,24 +31,46 @@
 #include "qspi.h"
 #include "pin_static_af.h"
 
-#if defined(MICROPY_HW_QSPIFLASH_SIZE_BITS_LOG2)
+#if defined(MICROPY_HW_QSPIFLASH_SIZE_BYTES) || defined(MICROPY_HW_QSPIFLASH_SIZE_BITS_LOG2)
 
 #ifndef MICROPY_HW_QSPI_PRESCALER
-#define MICROPY_HW_QSPI_PRESCALER       3  // F_CLK = F_AHB/3 (72MHz when CPU is 216MHz)
+#define MICROPY_HW_QSPI_PRESCALER            3  // F_CLK = F_AHB/3 (72MHz when CPU is 216MHz)
 #endif
 
 #ifndef MICROPY_HW_QSPI_SAMPLE_SHIFT
-#define MICROPY_HW_QSPI_SAMPLE_SHIFT    1  // sample shift enabled
+#define MICROPY_HW_QSPI_SAMPLE_SHIFT         1  // sample shift enabled
 #endif
 
 #ifndef MICROPY_HW_QSPI_TIMEOUT_COUNTER
-#define MICROPY_HW_QSPI_TIMEOUT_COUNTER 0  // timeout counter disabled (see F7 errata)
+#define MICROPY_HW_QSPI_TIMEOUT_COUNTER      0  // timeout counter disabled (see F7 errata)
 #endif
 
 #ifndef MICROPY_HW_QSPI_CS_HIGH_CYCLES
-#define MICROPY_HW_QSPI_CS_HIGH_CYCLES  2  // nCS stays high for 2 cycles
+#define MICROPY_HW_QSPI_CS_HIGH_CYCLES       2  // nCS stays high for 2 cycles
 #endif
 
+#ifndef MICROPY_HW_QSPI_ENABLE_MPU_CACHING
+#define MICROPY_HW_QSPI_ENABLE_MPU_CACHING   0 // MPU deny access by default
+#endif
+
+#ifndef MICROPY_HW_QSPI_ENABLE_MEMORY_MAPPED
+#define MICROPY_HW_QSPI_ENABLE_MEMORY_MAPPED 1 // Memory mapped mode enabled in idle
+#endif
+
+// Provides the MPU_REGION_SIZE_X value when passed the size of region in bytes
+// "m" must be a power of 2 between 32 and 4G (2**5 and 2**32) and this formula
+// computes the log2 of "m", minus 1
+#define MPU_REGION_SIZE(m) (((m) - 1) / (((m) - 1) % 255 + 1) / 255 % 255 * 8 + 7 - 86 / (((m) - 1) % 255 + 12) - 1)
```

Comment:
I'm happy to change to using BYTES instead of LOG2_BITS in the configuration.  In that case this macro should be reused from sdram.c.